Assets/Packages/com.studio23.ss2.puzzlesystem/Runtime/Script/Puzzles/CombinationPuzzle.cs

```diff
@@ -99,39 +104,71 @@ public void StartPuzzle()
             {
                 IsPuzzleStarted = true;// Invoke OnPuzzleStart 
                 SelectedDial = 0; // Fire OnSelectedDialChanged
-                PuzzleInfo.CheckPuzzleStatus(); // check if puzzle is solved
+                
+                if (PuzzleInfo.IsPuzzleSolved)
+                {
+                    OnPuzzleUnlocked.Invoke();
+                }
             }
             else
             {
                 Debug.Log("Puzzle is already started");
             }
         }
+       
+        public void Move(Direction direction)
+        {
+            if (!IsPuzzleStarted) 
+                return;
+            if (PuzzleInfo.IsPuzzleSolved) 
+                return;
+            
+            switch (direction)
```

Comment:
This looks wrong. Essentially this is doing 2 different operation. Left and Right is moving selection while up and down is rotating the dials. These two should be managed in separate functions.  Make two methods one called Select Dial and another called rotate dial. Pass in only floating values of direction x or y